### PR TITLE
fix: send spans also if secret masking is wrong [RD-2615]

### PR DESCRIPTION
**Pull request by doriaviram**

### CODE REVIEW

File: src/reporter.test.js [hunk 1]
```diff
@@ -102,6 +102,23 @@ describe('reporter', () => {
     expect(result.rtt).toBeGreaterThanOrEqual(0);
   });
 
+  test('sendSpans - with bad LUMIGO_SECRET_MASKING_REGEX still send spans', async () => {
+    const token = 'DEADBEEF';
+    process.env.LUMIGO_SECRET_MASKING_REGEX =
+      "['.*pass.*', '.*key.*', '.*secret.*', '.*credential.*', '.*passphrase.*']";
```
[the remark this answers is not in the document]
Comment (doriaviram) [reply src/reporter.test.js:108]: This is the test I wanted, to give an un-valid JSON. 
just make it more clear for u :)

File: src/utils.js [hunk 2]
```diff
@@ -364,12 +374,37 @@ export const shouldScrubDomain = url => {
...
+  if (process.env[LUMIGO_SECRET_MASKING_REGEX]) {
+    const parseResponse = parseJsonFromEnvVar(
+      LUMIGO_SECRET_MASKING_REGEX,
+      true
+    );
+    if (parseResponse) {
+      regexesList = parseResponse;
+    }
+  }
+  return regexesList.map(x => new RegExp(x, 'i'));
```
[the remark this answers is not in the document]
Comment (doriaviram) [reply src/utils.js:406]: Its copied for the old implementation, i is case-insensitive

File: src/utils.js [hunk 3]
```diff
@@ -364,12 +374,37 @@ export const shouldScrubDomain = url => {
...
+  }
+  if (process.env[LUMIGO_SECRET_MASKING_REGEX]) {
+    const parseResponse = parseJsonFromEnvVar(
+      LUMIGO_SECRET_MASKING_REGEX,
+      true
+    );
+    if (parseResponse) {
+      regexesList = parseResponse;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (doriaviram) [reply src/utils.js:405]: 👍

File: src/utils.js [hunk 4]
```diff
@@ -364,12 +374,38 @@ export const shouldScrubDomain = url => {
...
+  if (process.env[LUMIGO_SECRET_MASKING_REGEX_BACKWARD_COMP]) {
+    const parseResponse = parseJsonFromEnvVar(
+      LUMIGO_SECRET_MASKING_REGEX_BACKWARD_COMP,
+      true
+    );
+    if (parseResponse) {
+      regexesList = parseResponse;
+    }
+  } else {
+    if (process.env[LUMIGO_SECRET_MASKING_REGEX]) {
```
Comment (nirLumigo) [inline src/utils.js:397]: I meant `else if` like: 
image
:)

Comment (doriaviram) [reply src/utils.js]: Done

**Pull request merged**